src/main/scala/csv/Utils.scala

```diff
@@ -0,0 +1,112 @@
+package csv
+
+import java.io.File
+
+import csv.CrimeRecord._
+
+import scala.io.Source
+
+/**
+  *
+  * Utils to work with Crime Records Models
+  */
+object Utils {
+
+  /** Collect data from files:
+    * read files from folder,
+    * read lines from file,
+    * drop csv headers
+    *
+    * @param dir files location
+    * @return List[List[String]]
+    **/
+  def readLines(dir: String): List[List[String]] = {
+    val lines = new File(dir).listFiles.filter(_.getName.endsWith(".csv")).map(f => Source.fromFile(f).getLines().toList.drop(1)).toList
+    lines.flatMap(l => l.map(s => s.split(",").toList))
+  }
+
+  /**
+    * Compose Crime Records from csv lines
+    *
+    * @param lines csv
+    * @return List[CrimeRecord]
+    */
+  def compose(lines: List[List[String]]): List[CrimeRecord] = {
+    lines.map(l => build(l))
+  }
+
+  /**
+    * Filter Crime Records with empty Crime Id
+    *
+    * @param list Crime Records
+    * @return List[CrimeRecord]
+    */
+  def filterEmptyCrimeID(list: List[CrimeRecord]): List[CrimeRecord] = {
+    list.filter(e => e.crimeID != "?")
+  }
+
+  /**
+    * Group Crime Records by coordinates
+    *
+    * @param list Crime Records
+    * @return List[CrimeRecord]
+    */
+  def groupByCoordinates(list: List[CrimeRecord]): List[CrimeRecord] = {
+    list.groupBy(e => (e.longitude, e.latitude)).flatMap(t => t._2).toList
+  }
+
+  /**
+    * Add total number of crime per coordinates to Crime Records model and sort by total number of crime per coordinates
+    *
+    * @param list Crime Records
+    * @return List[CrimeRecord]
+    */
+  def sortByTotalNumberOfCrime(list: List[CrimeRecord]) = {
+    list.filterNot(cr => cr.location == "No Location").groupBy(e => (e.longitude, e.latitude)).flatMap {
+      t =>
+        val count = t._2.size
+        t._2.map(e => e.copy(13, count.toString))
+    }.toList.sortWith((cr1, cr2) => cr1.locationCrimeNumber.toInt > cr2.locationCrimeNumber.toInt).take(5)
```

Comment:
function body doesn't fully correspond with the name: it mutates records and returns only 5 records